server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -962,59 +970,73 @@ public Query fuzzyQuery(
+            return new ConstantScoreQuery(super.multiPhraseQuery(name, stream, slop, enablePositionIncrements, queryShardContext));
         }
 
         @Override
-        public Query phrasePrefixQuery(TokenStream stream, int slop, int maxExpansions, SearchExecutionContext queryShardContext)
-            throws IOException {
+        public Query phrasePrefixQuery(
+            String name,
+            TokenStream stream,
+            int slop,
+            int maxExpansions,
+            SearchExecutionContext queryShardContext
+        ) throws IOException {
             // Disable scoring
-            return new ConstantScoreQuery(super.phrasePrefixQuery(stream, slop, maxExpansions, queryShardContext));
+            return new ConstantScoreQuery(super.phrasePrefixQuery(name, stream, slop, maxExpansions, queryShardContext));
         }
 
     }
 
     static class LegacyTextFieldType extends ConstantScoreTextFieldType {
 
-        private final MappedFieldType existQueryFieldType;
-
-        LegacyTextFieldType(String name, boolean indexed, boolean stored, TextSearchInfo tsi, Map<String, String> meta) {
-            super(name, indexed, stored, tsi, meta);
-            // norms are not available, neither are doc-values, so fall back to _source to run exists query
-            existQueryFieldType = KeywordScriptFieldType.sourceOnly(name()).asMappedFieldTypes().findFirst().get();
+        LegacyTextFieldType(boolean indexed, boolean stored, TextSearchInfo tsi, Map<String, String> meta) {
+            super(indexed, stored, tsi, meta);
         }
 
         @Override
-        public SpanQuery spanPrefixQuery(String value, SpanMultiTermQueryWrapper.SpanRewriteMethod method, SearchExecutionContext context) {
-            throw new IllegalArgumentException("Cannot use span prefix queries on text field " + name() + " of a legacy index");
+        public SpanQuery spanPrefixQuery(
+            String name,
+            String value,
+            SpanMultiTermQueryWrapper.SpanRewriteMethod method,
+            SearchExecutionContext context
+        ) {
+            throw new IllegalArgumentException("Cannot use span prefix queries on text field " + name + " of a legacy index");
         }
 
         @Override
-        public Query existsQuery(SearchExecutionContext context) {
+        public Query existsQuery(String name, SearchExecutionContext context) {
             if (context.allowExpensiveQueries() == false) {
                 throw new ElasticsearchException(
                     "runtime-computed exists query cannot be executed while [" + ALLOW_EXPENSIVE_QUERIES.getKey() + "] is set to [false]."
                 );
             }
-            return existQueryFieldType.existsQuery(context);
+            // norms are not available, neither are doc-values, so fall back to _source to run exists query
+            final MappedField existQueryField = KeywordScriptFieldType.sourceOnly(name).asMappedFields().findFirst().get();
```

Comment:
this was moved from the constructor of LegacyTextFieldType now to the invocation site here, as we need the `name` to construct the existQueryField